client/client.go

```diff
@@ -61,6 +61,9 @@ func (c *genericClient) receive() (*pb.Message, error) {
 	} else if err != nil {
 		return nil, fmt.Errorf("[Client %v] An error ocurred: %v", c.id, err)
 	}
+	if resp.ErrorMsg != "" {
```

Comment:
Okay, you're right, I now see the need to differentiate gRPC-related errors from protocol execution errors. 

I would only suggest renaming `ErrorMsg` to `ProtocolError` in `msgs.proto` (requires change in L64 as well as in server-side functions for pseudonym system).

We have to remember to implement sending protocol error messages from other server-side functions of other protocols  in the near future (perhaps as a part of server-side refactoring).